File: WebRouterApp/ClientApp/src/app/trading/models/messages/orders-snapshot.message.ts
import { IOrder } from '../order';
import { TradingMessageTags } from './trading-message-tags';
import { Guid } from '../../../shared/types/guid';

export interface IOrdersSnapshotMessage {
  tag: TradingMessageTags.OrdersSnapshot;
  traderId: Guid;
  orders: IOrder[];
}

File: WebRouterApp/Features/Trading/Application/MessageParts/OrdersSnapshotMessage.cs
using System;
using System.Collections.Generic;

namespace WebRouterApp.Features.Trading.Application.MessageParts
{
    public class OrdersSnapshotMessage : ITradingMessage
    {
        public OrdersSnapshotMessage(Guid traderId, IReadOnlyList<OrderModel> orders)
        {
            TraderId = traderId;
            Orders = orders;
        }

        public string Tag => "Trading.OrdersSnapshot";

        public Guid TraderId { get; }
        public IReadOnlyList<OrderModel> Orders { get; }
    }
}